app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,41 @@
 class PostsController < ApplicationController
+  before_action :fetch_user, only: %i[index show]
+
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @posts = @user ? @user.posts : Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @user = current_user
+    @post.user = @user
+
+    if @post.save
+      flash[:notice] = 'Post was successfully created.'
+      redirect_to user_posts_path(@user)
+    else
+      redirect_to new_post_path
+    end
   end
```

Comment:
- [ ] I am unable to create a post. If you check the errors related to `@post` you would see that comments and likes counters are not defined so the validation failed. You need to implement both counters before attempting to save.

  ```
  @post.likes_counter = 0
  @post.comments_counter = 0
  ```

  ![image](https://user-images.githubusercontent.com/54848363/205399437-3b0573b9-2229-4d41-9c08-2a10b70d7028.png)

Reply:
Thank you for the feedback.